### ISSUE TITLE
tf.nn.selu returns wrong gradient for large negative float64 input

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#124834]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

 Linux Ubuntu 22.04

### Python version

Python 3.13

### Current behavior?

`tf.nn.selu` returns an incorrect gradient for a large negative float64 input.

For SELU, the negative branch is a scaled ELU branch. At `x = -700.0`, the analytic high-precision reference for the derivative is approximately `1.7334290832552396e-304`, which is a finite nonzero float64 value.

However, TensorFlow autodiff returns `0.0` for this gradient. PyTorch computes the expected finite value for the same derivative, so this appears to be a backward-pass issue rather than a float64 limitation.

Expected behavior?

The gradient of `tf.nn.selu` at `x = -700.0` should be close to the analytic reference value `1.7334290832552396e-304`, not `0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

x = -700.0
truth = 1.7334290832552396e-304
rtol = 1e-6

v = tf.Variable(x, dtype=tf.float64)

with tf.GradientTape() as tape:
    y = tf.nn.selu(v)

g = tape.gradient(y, v)

actual = float(g)
rel_err = abs(actual - truth) / abs(truth)

print(f"x: {x!r}")
print(f"tensorflow gradient: {actual!r}")
print(f"expected gradient: {truth!r}")
print(f"relative error: {rel_err:.3e}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.nn.selu gradient is wrong for large negative float64 input")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: -700.0
tensorflow gradient: 0.0
expected gradient: 1.7334290832552396e-304
relative error: 1.000e+00
BUG REPRODUCED: tf.nn.selu gradient is wrong for large negative float64 input
```

**Comment by GodlyDonuts** [comment]
I am working on a focused fix for this numerical gradient bug with regression coverage. I will keep this separate from the related ELU and higher-order-gradient issues.

**Comment by Kayyuri** [comment]
Hi @ALinrunrun, Thank you for raising this issue!
A [PR](https://github.com/tensorflow/tensorflow/pull/125171) has been submitted to address this issue and will be fixed once it is merged.

Many thanks to @GodlyDonuts for contributing a fix to TensorFlow.
